### PR TITLE
Feature/max statements with exclusions

### PR DESCRIPTION
**Pull request by ClareJolly**
Updated to use callback and tests passing.  Need to test it in a repo and add some more tests but wanted to raise a PR to check the cicd checks

### CODE REVIEW

File: src/rules/maxStatementsWithExclusions/mockCallback.js [hunk 1]
```diff
@@ -0,0 +1,15 @@
+const lodash = require('lodash');
+
+const mockCallback = (node) => {
+  try {
+    const test = node.body.map((n) => {
+      return lodash.get(n, 'expression.callee.object.name', undefined);
+    });
+    const filtered = test.filter((t) => t !== 'logger').length;
+    return filtered;
```
Comment (amirmohsen) [inline src/rules/maxStatementsWithExclusions/mockCallback.js]: Useless var

File: src/rules/maxStatementsWithExclusions/maxStatementsWithExclusions.test.js [hunk 2]
```diff
@@ -0,0 +1,183 @@
...
+  `function Test() {
+  console.log('1');
+  console.log('2');
+  console.log('3');
+  console.log('4');
+  console.log('5');
+  logger.log('6');
+}
+`,
+];
```
Comment (amirmohsen) [inline src/rules/maxStatementsWithExclusions/maxStatementsWithExclusions.test.js]: Extra test cases for both valid and invalid scenarios:
- function declaration
- function expression
- instance class methods
- static class methods
- constructor
- arrow function
- arrow function inside class (as a field)
- nested functions (combinations of the above)
- exported functions (variations of the above)

**Review by amirmohsen — CHANGES_REQUESTED** [review]
Could you extend the jest config to provide the `collectCoverageFrom` option in order to limit the scope to only the `src` folder and then move your mock out into a `test/unit/utils/` folder and rename and require from there.

Jest option: https://vfuk-digital.visualstudio.com/Digital/_workitems/edit/358135/

And fix the linting issue please.

File: src/rules/maxStatementsWithExclusions/mockCallback.js (hunk 1 above)
Comment (ClareJolly) [reply src/rules/maxStatementsWithExclusions/mockCallback.js]: this is done (in the file that's in the new location)

File: src/rules/maxStatementsWithExclusions/maxStatementsWithExclusions.test.js (hunk 2 above)
Comment (ClareJolly) [reply src/rules/maxStatementsWithExclusions/maxStatementsWithExclusions.test.js]: the above tests added - and some test helpers to produce the various function types - seems to work well - you make think it's overkill but I didn't want to have to keep writing out all the scenarios.  I think I made it reusable for other tests - at least for statements anyway

Linting run and all tests passing.

Please can you check and authorise to run the checks workflow

**Review by amirmohsen — APPROVED** [review]
Excellent work! I really like your composition of different test cases.

**Pull request merged**